### ISSUE TITLE
build error: No rule to make target 'scripts/escape'

### ISSUE THREAD

**Issue by wojdyr** [cod-developers/cod-tools#1]
Trying to build svn r4789, I do `make` and get at the end:
```
tests/cases/utf8-to-cif_003.inp: OK
tests/cases/utf8-to-cif_004.inp: OK
tests/shtests/cif_find_duplicates_006.sh: OK
make: *** No rule to make target 'scripts/escape', needed by 'tests/shoutputs/escape_001.diff'. Stop.
```

**Comment by sauliusg** [comment]
Dear Marcin,

thank you for your bug report and sorry for the delayed reply (we had some difficulties to reproduce the bug reliably...)

Please check the latest fix pushes (commit 6f42bcd, svn revision 4802), the cod-tools should built now at least on Debian-based systems. If the bug persists please let me know.

Kind regards,
Saulius

**Comment by sauliusg** [comment]
PS: you'll probably have to 'make distclean; make' to have the dependencies fixed.

**Comment by wojdyr** [comment]
Thanks! I'll try it at home. It doesn't work in my office because 'make' at some point wants
to connect to MySQL server on 'www.crystallography.net' and Mysql port is apparently blocked by our firewall.

**Comment by wojdyr** [comment]
`make` did make it to the end, although with multiple "FAILED" messages.
But I suppose it's only tests, the build part probably worked.
Some failures are due to failure to contact MySQL, but there are also others, a few of them I pasted below.
Anyway, the original problem seems to be fixed.
I'm running it on Ubuntu 16.04.

```
tests/cases/cif_find_duplicates_011.opt: OK
tests/cases/cif_find_symmetry_001.inp: FAILED:
52,53c52,53
< O9 O 1.14562922099e-17 0.7111 0.25 0.00000 Uiso 1.00
< O10 O 2.92652947418e-18 0.9262 0.25 0.00000 Uiso 1.00
---
> O9 O 1.1453911918e-17 0.7111 0.25 0.00000 Uiso 1.00
> O10 O 2.92592142454e-18 0.9262 0.25 0.00000 Uiso 1.00
tests/cases/cif_find_symmetry_002.inp: OK
tests/cases/cif_find_symmetry_003.inp: OK
tests/cases/cif_find_symmetry_004.inp: FAILED:
119c119
< Fe FE1 1 1 0.21021 0.46 Uiso 0.0053(4) 1
---
> Fe FE1 0 0 0.21021 0.46 Uiso 0.0053(4) 1
tests/cases/cif_find_symmetry_005.inp: OK
tests/cases/cif_find_symmetry_006.inp: FAILED:
39,43c39,43
< Cu 0 2.77555756156e-17 0.5
< Cu 0 3.46944695195e-18 0.94
< Cu 0 6.93889390391e-18 0.867
< Cu 0 1.38777878078e-17 0.75
< Cu 0 2.77555756156e-17 0.65
---
> Cu 0 1.11022302463e-16 0.5
> Cu 0 1.38777878078e-17 0.94
> Cu 0 2.77555756156e-17 0.867
> Cu 0 5.55111512313e-17 0.75
> Cu 0 8.32667268469e-17 0.65
45,46c45,46
< S 0 1.38777878078e-17 0.8
< S 0 2.77555756156e-17 0.6
---
> S 0 5.55111512313e-17 0.8
> S 0 1.11022302463e-16 0.6
tests/cases/cif_find_symmetry_007.inp: OK
tests/cases/cif_find_symmetry_008.inp: OK
tests/cases/cif_find_symmetry_009.inp: FAILED:
30,32c30,32
< Cl 0.802699946889 0.605010046724 0.512100207955
< Cl 0.197299932838 0.605010046724 0.487900490819
< Cl 0.979900151378 0.895460212554 0.177600465104
---
> Cl 0.80269994689 0.605010046724 0.512100207955
> Cl 0.197299932839 0.605010046724 0.487900490819
> Cl 0.979900151379 0.895460212554 0.177600465104
34c34
< Cl 2.06924499959e-07 0.208819933668 0.499999767075
---
> Cl 2.06924500021e-07 0.208819933668 0.499999767075
36,37c36,37
< K 0.109099596585 0.661209964413 0.822699541977
< Ta 0.635620424105 0.587456726463 0.641300074163
---
> K 0.109099596586 0.661209964413 0.822699541977
> Ta 0.635620424106 0.587456726463 0.641300074163
```

**Comment by sauliusg** [comment]
On 2017-01-25 18:53, Marcin Wojdyr wrote:
> Thanks! I'll try it at home. It doesn't work in my office because 'make'
> at some point wants
> to connect to MySQL server on 'www.crystallography.net' and Mysql port
> is apparently blocked by our firewall.
> 

Hi, thanks for the update!

You can ignore the mysql connection failures -- these check one script,
'cif_cod_numbers', and all the rest can be tested without the net (we'll
have to implement 'skip' for the networked tests if the connection is
not available, to avoid confusion).

Cheers,
Saulius

-- 
Dr. Saulius Gražulis
Vilnius University Institute of Biotechnology, Saulėtekio al. 7
LT-10223 Vilnius, Lietuva (Lithuania)
fax: (+370-5)-2234367 / phone (office): (+370-5)-2234353
mobile: (+370-684)-49802, (+370-614)-36366

**Comment by sauliusg** [comment]
On 2017-01-25 19:40, Marcin Wojdyr wrote:
> |make| did make it to the end, although with multiple "FAILED" messages.
> But I suppose it's only tests, the build part probably worked.

You are right, "FAILED" indicates a failed test, and the test is
assessed by diff-ing the tested program's outputs with the expected ones
(recorded on the *.out files under the test directory). You can run
'make listdiff' to get a list of failed tests, BTW.

The list you have sent me is clearly related to FP rounding problems.
You probably run a 64 bit version, and our tests were set up on a 32-bit
host, so we still have problems migrating to multiple environments.

If you could send me the output of your 'uname -a' and 'make listdiff'
in your cod-tools directory, this would be very helpful for us to adapt
out tests so that they do not fail spuriously.

The bottom line: the cod-tools that you have compiled should work
correctly on your machine; expect small but insignificant textual
differences in outputs of floating point numbers.

Regards,
Saulius

-- 
Dr. Saulius Gražulis
Vilnius University Institute of Biotechnology, Saulėtekio al. 7
LT-10223 Vilnius, Lietuva (Lithuania)
fax: (+370-5)-2234367 / phone (office): (+370-5)-2234353
mobile: (+370-684)-49802, (+370-614)-36366

**Comment by sauliusg** [comment]
On 2017-01-25 19:40, Marcin Wojdyr wrote:
> Closed #1 <https://github.com/sauliusg/cod-tools/issues/1>.
> 

Thanks!

-- 
Dr. Saulius Gražulis
Vilnius University Institute of Biotechnology, Saulėtekio al. 7
LT-10223 Vilnius, Lietuva (Lithuania)
fax: (+370-5)-2234367 / phone (office): (+370-5)-2234353
mobile: (+370-684)-49802, (+370-614)-36366

**Comment by wojdyr** [comment]
```
$ uname -a
Linux el 4.4.0-59-generic #80-Ubuntu SMP Fri Jan 6 17:47:47 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

$ make listdiff
-rw-rw-r-- 1 wojdyr wojdyr    915 Jan 25 16:50 ./tests/outputs/cif_cod_numbers_001.diff
-rw-rw-r-- 1 wojdyr wojdyr    664 Jan 25 16:52 ./tests/outputs/cif_cod_numbers_002.diff
-rw-rw-r-- 1 wojdyr wojdyr    832 Jan 25 16:54 ./tests/outputs/cif_cod_numbers_003.diff
-rw-rw-r-- 1 wojdyr wojdyr    664 Jan 25 16:56 ./tests/outputs/cif_cod_numbers_004.diff
-rw-rw-r-- 1 wojdyr wojdyr    664 Jan 25 16:58 ./tests/outputs/cif_cod_numbers_005.diff
-rw-rw-r-- 1 wojdyr wojdyr    915 Jan 25 17:01 ./tests/outputs/cif_cod_numbers_006.diff
-rw-rw-r-- 1 wojdyr wojdyr   1042 Jan 25 17:03 ./tests/outputs/cif_cod_numbers_007.diff
-rw-rw-r-- 1 wojdyr wojdyr    804 Jan 25 17:05 ./tests/outputs/cif_cod_numbers_009.diff
-rw-rw-r-- 1 wojdyr wojdyr    708 Jan 25 17:07 ./tests/outputs/cif_cod_numbers_010.diff
-rw-rw-r-- 1 wojdyr wojdyr    898 Jan 25 17:09 ./tests/outputs/cif_cod_numbers_011.diff
-rw-rw-r-- 1 wojdyr wojdyr    804 Jan 25 17:11 ./tests/outputs/cif_cod_numbers_012.diff
-rw-rw-r-- 1 wojdyr wojdyr    708 Jan 25 17:13 ./tests/outputs/cif_cod_numbers_013.diff
-rw-rw-r-- 1 wojdyr wojdyr    708 Jan 25 17:15 ./tests/outputs/cif_cod_numbers_014.diff
-rw-rw-r-- 1 wojdyr wojdyr   1910 Jan 25 17:18 ./tests/outputs/cif_cod_numbers_015.diff
-rw-rw-r-- 1 wojdyr wojdyr    836 Jan 25 17:20 ./tests/outputs/cif_cod_numbers_016.diff
-rw-rw-r-- 1 wojdyr wojdyr    988 Jan 25 17:22 ./tests/outputs/cif_cod_numbers_017.diff
-rw-rw-r-- 1 wojdyr wojdyr    804 Jan 25 17:24 ./tests/outputs/cif_cod_numbers_018.diff
-rw-rw-r-- 1 wojdyr wojdyr    237 Jan 25 17:25 ./tests/outputs/cif_find_symmetry_001.diff
-rw-rw-r-- 1 wojdyr wojdyr     98 Jan 25 17:25 ./tests/outputs/cif_find_symmetry_004.diff
-rw-rw-r-- 1 wojdyr wojdyr    444 Jan 25 17:25 ./tests/outputs/cif_find_symmetry_006.diff
-rw-rw-r-- 1 wojdyr wojdyr    645 Jan 25 17:25 ./tests/outputs/cif_find_symmetry_009.diff
-rw-rw-r-- 1 wojdyr wojdyr    588 Jan 25 17:27 ./tests/outputs/cod_predeposition_check_001.diff
-rw-rw-r-- 1 wojdyr wojdyr  87341 Jan 25 17:27 ./tests/outputs/cod_predeposition_check_002.diff
-rw-rw-r-- 1 wojdyr wojdyr    636 Jan 25 17:27 ./tests/outputs/cod_predeposition_check_028.diff
-rw-rw-r-- 1 wojdyr wojdyr    687 Jan 25 17:27 ./tests/outputs/cod_predeposition_check_029.diff
-rw-rw-r-- 1 wojdyr wojdyr    679 Jan 25 17:27 ./tests/outputs/cod_predeposition_check_030.diff
-rw-rw-r-- 1 wojdyr wojdyr 153686 Jan 25 17:27 ./tests/outputs/cod_predeposition_check_032.diff
-rw-rw-r-- 1 wojdyr wojdyr    572 Jan 25 17:30 ./tests/shoutputs/cif_cod_numbers_001.diff
-rw-rw-r-- 1 wojdyr wojdyr    572 Jan 25 17:32 ./tests/shoutputs/cif_cod_numbers_002.diff
```

**Comment by sauliusg** [comment]
On 2017-01-26 15:55, Marcin Wojdyr wrote:
> |$ uname -a Linux el 4.4.0-59-generic #80-Ubuntu SMP Fri Jan 6 17:47:47
> UTC 2017 x86_64 x86_64 x86_64 GNU/Linux $ make listdiff -rw-rw-r-- 1
> wojdyr wojdyr 915 Jan 25 16:50 ./tests/outputs/cif_cod_numbers_001.diff
> -rw-rw-r-- 1 wojdyr wojdyr 664 Jan 25 16:52
> ./tests/outputs/cif_cod_numbers_002.diff -rw-rw-r-- 1 wojdyr wojdyr 832

Thank you, we will investigate!

Cheers,
Saulius

-- 
Dr. Saulius Gražulis
Vilnius University Institute of Biotechnology, Saulėtekio al. 7
LT-10223 Vilnius, Lietuva (Lithuania)
fax: (+370-5)-2234367 / phone (office): (+370-5)-2234353
mobile: (+370-684)-49802, (+370-614)-36366
